src/org/jetbrains/plugins/scala/settings/HydraApplicationSettings.scala

```diff
@@ -0,0 +1,62 @@
+package org.jetbrains.plugins.scala.settings
+
+import java.io.File
+import java.util
+import scala.collection.JavaConverters._
+import com.intellij.openapi.components._
+
+import scala.beans.BeanProperty
+
+/**
+  * @author Maris Alexandru
+  */
+@State(
+  name = "HydraApplicationSettings",
+  storages = Array(new Storage("hydra_config.xml"))
+)
+class HydraApplicationSettings extends PersistentStateComponent[HydraApplicationSettingsState]{
+
+  var artifactPaths: Map[(String, String), List[String]] = Map.empty
```

Comment:
This used to be a `ConcurrentHashMap` before (named `cahe`) but it's now mutable (not the map, but the reference to the map). I'm wondering if this field is going to be accessed from different threads. Should this class be thread-safe?